test/composition.jl

```diff
@@ -14,7 +17,11 @@ using Test, LinearMaps, LinearAlgebra
     @test @inferred (F * A) * v == @inferred F * (A * v)
     @test @inferred (A * F) * v == @inferred A * (F * v)
     @test @inferred A * (F * F) * v == @inferred A * (F * (F * v))
-    @test @inferred (F * F) * (F * F) * v == @inferred F * (F * (F * (F * v)))
+    F2 = F*F
+    FC2 = FC*FC
+    F4 = FC2 * F2
+    @test length(F4.maps) == 4
+    @test @inferred F4 * v == @inferred F * (F * (F * (F * v)))
```

Comment:
i'm surprised that no `isapprox` is needed here but apparently it works...

Reply:
This is because this is just straightforward addition of the vector elements (`cumsum`). There is no issue with associativity in multiply-and-add, because there is no multiplication involved. I mean `w[k] = w[k-1]+v[k]`, not `w[k] = 1*w[k-1]+1*v[k]` or anything like that, as you would get from representing `cumsum` as a matrix.